### ISSUE TITLE
Using open-llama instead of LLaMA

### ISSUE THREAD

**Issue by kstevica** [Lightning-AI/lit-llama#224]
I have tried to convert open-llama from https://huggingface.co/openlm-research/open_llama_7b_preview_200bt/tree/main, converting from transformers version to "normal" using convert_hf_checkpoint and then convert_checkpoint, it failed in later one at:

```
  converted["transformer.wte.weight"] = state_dict["tok_embeddings.weight"].to(dtype)
  converted["lm_head.weight"] = state_dict["output.weight"].to(dtype)
  converted["transformer.ln_f.scale"] = state_dict["norm.weight"].to(dtype)
``` 

Keys tok_embeddings.weight, output.weight and norm.weight are missing in state_dict. I tried avoiding that stating

`converted = state_dict`

but I got nothing, inference was broken. I am pretty much sure I was shooting in a dark with an empty gun. Any help is appreciated.

**Comment by carmocca** [comment]
OpenLLaMA used https://github.com/young-geng/EasyLM to train their checkpoint. It's likely that there's a mismatch in the weights. We would need a specific conversion for it.

The current `convert_hf_checkpoint` code is tailored to weights generated by the original LLaMA implementation from https://github.com/facebookresearch/llama

**Comment by vihangd** [comment]
Seems to work fine with the latest weights with 300bt 

> $ python generate.py --prompt "what is the meaning of life?"
> Loading model ...
> Time to load model: 22.46 seconds.
> Global seed set to 1234
> what is the meaning of life? What is the meaning of life? What does it mean to be happy? How do I choose my life goals? What makes a good life? How do I become an achievement-oriented
> person? What is the secret to happiness? How do I achieve
> Time for inference 1: 3.50 sec total, 14.27 tokens/sec
> Memory used: 13.52 GB

**Comment by lantiga** [comment]
Oh nice! Can you confirm the steps you ran?

**Comment by vihangd** [comment]
@lantiga I just copied the transformer weights are ran the convert script

    python scripts/convert_hf_checkpoint.py --model_size 7B

**Comment by lantiga** [comment]
Awesome, thank you! We'll add to the howto and feature this in the README
